sdk/opa.go

```diff
@@ -272,7 +272,7 @@ func newDecisionResult() (*DecisionResult, error) {
 	return result, nil
 }
 
-func (opa *OPA) executeTransaction(ctx context.Context, record *server.Info, work func(result *DecisionResult)) (*DecisionResult, error) {
+func (opa *OPA) executeTransaction(ctx context.Context, record *server.Info, work func(state state, result *DecisionResult)) (*DecisionResult, error) {
```

Comment:
[nit] to avoid confusion
```suggestion
func (opa *OPA) executeTransaction(ctx context.Context, record *server.Info, work func(state, *DecisionResult)) (*DecisionResult, error) {
```